src/jsonpatcherproxy.js

```diff
@@ -76,8 +85,30 @@ var JSONPatcherProxy = (function() {
         return Reflect.get(target, propKey, receiver);
       },
       set: function(target, key, receiver) {
-        const distPath =
-          path + '/' + JSONPatcherProxy.escapePathComponent(key.toString());
+        var distPath;
+        var cachedProxy;
+
+        /* each proxified object has its path cached, we need to use that instead of `path` variable
+        because at one point in the future, paths might change and we will simply update our cache instead of 
+        proxifying again.  */
+        if ((cachedProxy = instance.objectsPathsMap.get(target))) {
```

Comment:
I find this notation very incomprehensible. Can this be changed to:

```js
cachedProxy = instance.objectsPathsMap.get(target);
if (cachedProxy) {
```

In general, it would be nice to have a weak map helper function `getOrSetPath(target, path)` that returns the path. Then you can get rid of this if/else here.